sdk/eventgrid/Azure.Messaging.EventGrid/src/Customization/EventGridModelFactory.cs

```diff
@@ -180,6 +181,15 @@ public static ResourceActionCancelEventData ResourceActionCancelEventData(string
         {
             return new(tenantId, subscriptionId, resourceGroup, resourceProvider, resourceUri, operationName, status, JsonDocument.Parse(authorization).RootElement, JsonDocument.Parse(claims).RootElement, correlationId, JsonDocument.Parse(httpRequest).RootElement);
         }
+
+        /// <summary> Initializes new instance of SubscriptionValidationResponse class. </summary>
+        /// <param name="validationResponse"> The validation response sent by the subscriber to Azure Event Grid to complete the validation of an event subscription. </param>
+        /// <returns> A new <see cref="SystemEvents.SubscriptionValidationResponse"/> instance for mocking. </returns>
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public static SubscriptionValidationResponse SubscriptionValidationResponse(string validationResponse = default)
```

Comment:
Do we need to hide the factory?  I get that you can use the constructor directly but it feels nice to be able to create any model from the model factory.

Reply:
We don't need to, but it would go against our pattern of only including factories for output models. Codegen no longer generates the model factory for this type, so I had to add it back manually.

Reply:
Yes - the model factory only exists for constructing the otherwise unconstructible per the guidelines today:  https://azure.github.io/azure-sdk/dotnet_introduction.html#dotnet-mocking-factory-builder

It would take nontrivial effort to design and implement this across all of Track 2 when there's already a way of accomplishing it now, so it's not a high priority to me.